mongo/filterstore.go

```diff
@@ -72,12 +76,15 @@ func (s *FilterStore) UpdateFilter(updatedFilter *models.Filter) error {
 	session := s.Session.Copy()
 	defer session.Close()
 
-	update := createUpdateFilterBlueprint(updatedFilter, time.Now())
+	update, err := mongolib.WithUpdates(createUpdateFilterBlueprint(updatedFilter, time.Now()))
```

Comment:
Should not be passing in time.Now() as all timestamps are handled by `mongolib.WithUpdates`.

This means removing all instances within mongo filestore.go where the `last_updated` field gets setOnInsert as we should always be relying on the `mongolib.WithUpdates` function to handle this for us.